### PR TITLE
Improvements of automatix daemon : Closes #5616, #5617

### LINKED ISSUE #5616: Use uploadclient in automatix
Motivation
----------
A specific `upload` method was implemented in `automatix` before `uploadclient` was developed. It is not needed anymore, and `uploadclient` should be used in the daemon now.

### LINKED ISSUE #5617: Heartbeat handler for automatix daemon
Motivation
----------
Heartbeat handler for automatix daemon (see https://github.com/rucio/rucio/issues/5562)

### PR DESCRIPTION
**Pull request by cserf**
Improvements of automatix daemon :
- Implementation of heartbeat handler : Closes #5617 
- Use uploadclient instead of upload method previously implemented in automatix : Closes #5616 
- Add unit tests

### CODE REVIEW

**Comment by rcarpa** [comment]
This is a nightmare to review. Please don't mix multiple unrelated changes in the same commit anymore. Even if I spend an hour trying to understand every single change, I'm still sure I'll miss a bug if there is any. 

Moving code around should ideally go into one commit. And adding new functionality to another commit. And, even if the previous ones are mixed, applying code formater should _definitely_! go into a separate commit.

File: lib/rucio/daemons/automatix/automatix.py [hunk 1]
```diff
@@ -210,58 +139,61 @@ def automatix(sites, inputfile, sleep_time, account, worker_number=1, total_work
         logger(logging.INFO, 'Getting data distribution')
         probabilities, data = get_data_distribution(inputfile)
         logger(logging.DEBUG, 'Probabilities %s', probabilities)
-        totretries = 3
```
Comment (rcarpa) [inline lib/rucio/daemons/automatix/automatix.py:211]: why do we remove the retry mechanism? There is no explaination about it in the commit message.

File: lib/rucio/daemons/automatix/automatix.py [hunk 2]
```diff
@@ -280,9 +212,15 @@ def run(total_workers=1, once=False, inputfile=None, sleep_time=-1):
         raise exception.DatabaseException('Database was not updated, daemon won\'t start')
 
     try:
-        sites = [s.strip() for s in config_get('automatix', 'sites').split(',')]
+        rses = [s.strip() for s in config_get('automatix', 'rses').split(',')]  # TODO use config_get_list
```
Comment (rcarpa) [inline lib/rucio/daemons/automatix/automatix.py]: I don't understand why this change is bundled into the commit "Use uploadclient in automatix" . how is it related to the uploadclient usage ? Why do we need this two-level 'rses', then 'sites' config parsing? Is it a new functionality to prefer the "rses" naming over "sites". If yes, why doesn't it come in a separate PR (or, at least, commit) ?

File: lib/rucio/daemons/automatix/automatix.py [hunk 3]
```diff
@@ -68,210 +69,216 @@ def choose_element(probabilities: dict, data: str) -> float:
...
-        if once is True:
-            logger(logging.INFO, 'Run with once mode. Exiting')
-            break
-        tottime = time() - starttime
-        if status:
-            logger(logging.INFO, 'It took %s seconds to upload one dataset on %s', str(tottime), str(rses))
-            daemon_sleep(start_time=starttime, sleep_time=sleep_time, graceful_stop=GRACEFUL_STOP, logger=logger)
+def run_once(heartbeat_handler: "HeartbeatHandler", inputfile: str, **_kwargs) -> bool:
+
+    _, _, logger = heartbeat_handler.live()
```
Comment (rcarpa) [inline lib/rucio/daemons/automatix/automatix.py:134]: Why move this code from run() to run_once() in a commit called "Heartbeat handler for automatix daemon". If this is needed for this commit, it must be explained in the commit message. If it's unrelated, it must be in a different commit.

File: lib/rucio/daemons/automatix/automatix.py [hunk 4]
```diff
@@ -68,210 +69,216 @@ def choose_element(probabilities: dict, data: str) -> float:
...
+            remove(physical_fname)
+        rmdir(tmpdir)
+    tottime = time.time() - starttime
+    logger(
+        logging.INFO,
+        "It took %s seconds to upload one dataset on %s",
+        str(tottime),
+        str(rses),
+    )
+    return True
```
Comment (rcarpa) [inline lib/rucio/daemons/automatix/automatix.py:248]: feels like it will always sleep now, which wasn't the case before. Am I wrong?

File: lib/rucio/tests/test_automatix.py [hunk 5]
```diff
@@ -81,3 +84,5 @@ def test_automatix(vo, root_account, rse_factory):
         assert status
         for file_ in files:
             assert file_dict["%s:%s" % (file_["scope"], file_["name"])] is True
+    config_remove_option("automatix", "rses")
```
Comment (rcarpa) [inline lib/rucio/tests/test_automatix.py:87]: did you try using the `file_config_mock` fixture for this ? If the automatix test doesn't interact using the API with the rest server, it should be a better solution than marking the test noparallel.

Comment (cserf) [reply lib/rucio/tests/test_automatix.py:87]: There are 2 issues here :
- Some options depends on variables (e.g. rse coming from rse_factory) that are generated by the test itself
- Other options are conditional, based on environment variable (POLICY) and I don't think that the `file_config_mock` allows to do it

File: lib/rucio/daemons/automatix/automatix.py (hunk 1 above)
Comment (cserf) [reply lib/rucio/daemons/automatix/automatix.py:211]: I checked during the last month and this retry mechanism was never used. That's why it was removed

File: lib/rucio/daemons/automatix/automatix.py (hunk 2 above)
Comment (cserf) [reply lib/rucio/daemons/automatix/automatix.py]: Yes, that's new functionality that should have gone into an independent commit

File: lib/rucio/daemons/automatix/automatix.py (hunk 3 above)
Comment (cserf) [reply lib/rucio/daemons/automatix/automatix.py:134]: Main reason is that the config was resolved once at the start of the daemon, but then never refreshed

File: lib/rucio/daemons/automatix/automatix.py [hunk 6]
```diff
@@ -13,134 +13,52 @@
...
 import tempfile
 import threading
 from configparser import NoOptionError, NoSectionError
 from datetime import datetime
 from json import load
-from math import exp
-from os import remove, rmdir, stat, getpid
-from time import sleep, time
+from os import remove, rmdir
+import time
```
Comment (bari12) [inline lib/rucio/daemons/automatix/automatix.py]: Move import up

**Review by bari12 — COMMENTED** [review]
Other than Radus comments, One minor change + rebase please.

**Review by bari12 — APPROVED** [review]
